### PR TITLE
mutate: add a roadmap

### PR DESCRIPTION
**Pull request by joakim-brannstrom**

### CODE REVIEW

**Comment by joakim-brannstrom** [comment]
@peatersson @christoffer-nylen work in progress for now. I have only gotten time to start restructuring the texts. I'll start breaking down in a clearer roadmap.

File: plugin/mutate/README_roadmap.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+   implementation detail for the module
+ * the architecture should not limit the programing languages that can be
+   mutation tested. It should be possible to use the architecture for mutation
+   testing of any programming language
+ * enable reuse of components. A visualization component should be reusable
+   independent of the programming language that is mutation tested
+ * it should be possible to incrementally use mutation testing during development.
+ * it should be robust to infrastructure failures
+
+The purpose of this plugin is to perform mutation testing of a _target_.
```
Comment (christoffer-nylen) [inline plugin/mutate/README_roadmap.md:24]: What does this mean? What is a "target" ?

File: plugin/mutate/README_roadmap.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+The focus of this plugin is to use mutation testing to automate whole or part
+of the manual inspection of test procedures that are required when developing
+software to DAL-A and C according to RTCA-178C.
```
Comment (christoffer-nylen) [inline plugin/mutate/README_roadmap.md:28]: Suggestion: I think this should be removed.

File: plugin/mutate/README_roadmap.md [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+   coverage of a SUT. Could it also be used to "guide" us in what mutants to
+   prioritize?
+ * build a dependency for a file that is mutated such that it is only
+   re-analyzed if that file or any of its dependencies has changed. Use the
+   includeVisitor to find the dependencies.
+ * add a database query that returns test cases sorted by the number of mutants they killed.
+   change package.d to using it instead of `sort_tcs_on_kills`.
+ * rank mutants by how much they changed the source code. more tokens then
+   higher risk? Add to high interest mutants.  an alternative would be how much
+   they change the dataflow based on the LLVM IR.
```
Comment (christoffer-nylen) [inline plugin/mutate/README_roadmap.md:86]:
```suggestion
  * add statement coverage.
  * add an option to rank mutants by how much they change the code coverage (test impact).
  * rank mutants by how much they changed the source code. more tokens then
   higher risk? Add to high interest mutants.  an alternative would be how much
   they change the dataflow based on the LLVM IR.
```

File: plugin/mutate/README_roadmap.md (hunk 2 above)
Comment (joakim-brannstrom) [reply plugin/mutate/README_roadmap.md:28]: it is part of the purpose but it should be reformulated because it is generic

**Pull request merged**